### PR TITLE
Ports TG's/Tau Ceti's welding effect

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

Title.
Ports the TG port of Pariah Station's port of Tau Ceti's welding effect, adds support for it to many objects.
Because we have multi-tile objects I had to slightly alter the logic so the welding flame looks neater when applied to large doors.

## Why It's Good For The Game
It looks really neat and gives good visual feedback for when somebody is welding an object.
This example gif is from the TG codebase, but the same effect applies here if a marine welds a wall.
165455987-2917c11a-14e2-4a4e-adf6-50bd9d4e212c

## Changelog
:cl:
add: Ported TG's welder effect for use when welding doors/objects.
/:cl:

### CODE REVIEW

**Comment by novaepee** [comment]
Really good vfx to show that engineers are fixing cades.

**Comment by Nanu308** [comment]
Put Ckey next to the CL at the top, so people can see who added it ingame

**Comment by chesse20** [comment]
can we recolor this for the plasmacutter

File: code/game/objects/items/tools/maintenance_tools.dm [hunk 1]
```diff
@@ -199,7 +199,22 @@
...
+	if(istype(target, /obj/machinery/door/airlock/multi_tile))
+		if((target.dir == 1) || (target.dir == 2))
+			target.add_overlay(GLOB.welding_sparks_multitiledoor_vertical)
+		else
+			target.add_overlay(GLOB.welding_sparks_multitiledoor_horizontal)
+	else
+		target.add_overlay(GLOB.welding_sparks)
+	. = ..()
+	if(istype(target, /obj/machinery/door/airlock/multi_tile))
+		if((target.dir == 1) || (target.dir == 2))
```
Comment (TiviPlus) [inline code/game/objects/items/tools/maintenance_tools.dm:212]: A) dirs should use proper defines (ex: 1 = NORTH)
B) why not use & (ie thing.dir & NORTH)

File: code/game/turfs/walls/walls.dm [hunk 2]
```diff
@@ -338,12 +338,15 @@
 
 		user.visible_message(span_notice("[user] starts repairing the damage to [src]."),
 		span_notice("You start repairing the damage to [src]."))
+		src.add_overlay(GLOB.welding_sparks)
 		playsound(src, 'sound/items/welder.ogg', 25, 1)
```
Comment (TiviPlus) [inline code/game/turfs/walls/walls.dm:342]: remove all src.proc() in this file you added

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by Losenis** [comment]
Finally, now people will understand why I'm actually slugging them when I'm next to the cades getting shuffled

**Review by Kuro020 — APPROVED** [review]

**Comment by Losenis** [comment]
Wait does this also apply to repairing combat robots and yourself? Please say yes

**Comment by Lumipharon** [comment]
The only way this could be cooler is if you could move the weld effect while welding (i.e. make it follow the crack of an airlock, or randomly move spots when fixing shit)

**Comment by RipGrayson** [comment]
> Wait does this also apply to repairing combat robots and yourself? Please say yes

As of the most recent commit the welding effect will show when repairing metal limbs, but it's vaguely glitchy and I need to bugfix it more.

> The only way this could be cooler is if you could move the weld effect while welding (i.e. make it follow the crack of an airlock, or randomly move spots when fixing shit)

There's no ability to randomly adjust x or y coordinates for overlay effects that I'm aware of. If there were I wouldn't have to make a whole new effect just to get the welding flame to properly center on multi-tile doors

**Comment by Lumipharon** [comment]
> There's no ability to randomly adjust x or y coordinates for overlay effects that I'm aware of. If there were I wouldn't have to make a whole new effect just to get the welding flame to properly center on multi-tile doors

Dunno if it would actually be practical, but you can do it. 
You can check out gun attachment codes for x and y offsets and how it works there, then you could just be x_offset = rand (-5, 5), update_overlay etc. 

Then for welding airlocks closed you could have a different overlay effect, where the effect moves during the animation to follow where the airlock opening is. Actually come to think of it, that would also probs be easier for the randomisation above as well. Just make the overlay animation more frames, and have it move around in the animation itself. Of course that means you might need to have different animations for different objects.

If you don't want to mess with in this PR though, I can play with it after it's merged.

**Comment by MereWalker** [comment]
It's the small things that make the difference. Thank you for this!

File: code/game/objects/items/tools/maintenance_tools.dm [hunk 3]
```diff
@@ -199,7 +199,32 @@
 			var/mob/living/L = O
 			L.IgniteMob()
 
-
+/obj/item/tool/weldingtool/use_tool(atom/target, mob/living/user, delay, amount, volume, datum/callback/extra_checks)
+	if(istype(target, /obj/machinery/door/airlock/mainship/marine))
+		if((target.dir & NORTH) || (target.dir & SOUTH))
```
Comment (BraveMole) [inline code/game/objects/items/tools/maintenance_tools.dm:204]:
```suggestion
		if((target.dir & NORTH|SOUTH))
```
Should work, not tested

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**